modules/agar/src/main/ts/ephox/agar/api/Chain.ts

```diff
@@ -213,6 +215,23 @@ const predicate = <T>(p: (value: T) => boolean): Chain<T, T> =>
   on((input, next, die, logs) =>
     p(input) ? next(input, logs) : die('predicate did not succeed', logs));
 
+const toPromise = <A, B>(c: Chain<A, B>) => (a: A): Promise<B> =>
```

Comment:
So the `toPromise(c)(a)` is to make it easier to distinct between the chain and the value instead of using `toPromise(c, a)` or what there another reason?

Reply:
Good question... this is actually using the code from the spike @techtangents did, so going to defer to him on that one. I assume it's to make it easier to compose multiple promises once converted from a chain.

Reply:
I think I was expecting the `toPromise` call to be done in Agar in the libraries of actions. So, you'd have:

```
const cBlah: Chain<A, B> = ...

const pBlah: (a: A) => Promise<B> = toPromise(cBlah);
```

Then the test code would do:

```
it('does the needful', () => {

  const x = await pBlah(y);

});
```

So, while we change over, we'd have both `Chain` and `a -> Promise b` variants of each of these actions. It was only to help during changeover. 

Equally, you could change everything over to Promises, and have functions that turn Promises into Chains.